src/topp/MascotAdapter.cpp

```diff
@@ -487,10 +489,35 @@ class TOPPMascotAdapter :
 
     if (!mascot_out)
     {
+    // determine number of chunks of size n
+    int size = (experiment.size() - 1) / n_chunks + 1;
+ 
+    // create array of vectors to store the chunks
+    std::vector<PeakMap> split_experiment[size];
+
+    for (int k = 0; k < size; ++k)
+    {
+        // get range for next set of n elements
+        auto start_itr = std::next(experiment.cbegin(), k*n_chunks);
+        auto end_itr = std::next(experiment.cbegin(), k*n_chunks + n_chunks);
+
+        // allocate memory for the chunk
+        split_experiment[k].resize(n_chunks);
```

Comment:
this is too much for the last bin, in most cases.

you can resize for all bins, based on `.resize(end_itr - start_itr);`

Reply:
alright, thx i'll take a look at it next week again

Reply:
ok so I thought the if clause would be fine to resize the last chunk. If I do it instead with the `end_itr - start_itr`. Can I use the same iterators twice?